core/types/l2trace.go

```diff
@@ -0,0 +1,165 @@
+package types
+
+import (
+	"io"
+	"sort"
+	"strings"
+
+	"github.com/scroll-tech/go-ethereum/rlp"
+)
+
+// BlockResult returns block execute result for rollers.
+type BlockResult struct {
+	ExecutionResults []*ExecutionResult `json:"executionResults"`
+}
+
+type rlpBlockResult struct {
+	ExecutionResults []*ExecutionResult `json:"executionResults"`
+}
+
+func (b *BlockResult) EncodeRLP(w io.Writer) error {
+	return rlp.Encode(w, &rlpBlockResult{
+		ExecutionResults: b.ExecutionResults,
+	})
+}
+
+func (b *BlockResult) DecodeRLP(s *rlp.Stream) error {
+	var dec rlpBlockResult
+	err := s.Decode(&dec)
+	if err == nil {
+		b.ExecutionResults = dec.ExecutionResults
+	}
+	return err
+}
+
+// ExecutionResult groups all structured logs emitted by the EVM
+// while replaying a transaction in debug mode as well as transaction
+// execution status, the amount of gas used and the return value
+type ExecutionResult struct {
+	Gas         uint64         `json:"gas"`
+	Failed      bool           `json:"failed"`
+	ReturnValue string         `json:"returnValue,omitempty"`
+	StructLogs  []StructLogRes `json:"structLogs"`
+}
+
+type rlpExecutionResult struct {
+	Gas         uint64
+	Failed      bool
+	ReturnValue string `json:"returnValue,omitempty"`
```

Comment:
Do we need the json encoding for rlp struct? If so, add to other fields as well?

Reply:
Actually we don't need, this‘s redundant and can be deleted.